telegram/ext/jobqueue.py

```diff
@@ -348,42 +348,33 @@ def run_monthly(
         name = name or callback.__name__
         job = Job(callback, context, name, self)
 
-        if day_is_strict:
+        # if day == -1 the run on last day of the month
+        if day == -1:
```

Comment:
I think we can drop the big if else by using a one liner: `day='last' if day == -1 else day`